spyder/preferences/shortcuts.py

```diff
@@ -381,99 +423,62 @@ def update_warning(self, warning_type=NO_WARNING, conflicts=[]):
-            self.button_ok.setEnabled(True)
-            different_sequence = True
-
-        if sys.platform == 'darwin':
-            if 'Meta+Ctrl' in sequence:
-                shown_sequence = sequence.replace('Meta+Ctrl', 'Ctrl+Cmd')
-            elif 'Ctrl+Meta' in sequence:
-                shown_sequence = sequence.replace('Ctrl+Meta', 'Cmd+Ctrl')
-            elif 'Ctrl' in sequence:
-                shown_sequence = sequence.replace('Ctrl', 'Cmd')
-            elif 'Meta' in sequence:
-                shown_sequence = sequence.replace('Meta', 'Ctrl')
+            conflicts = self.check_conflicts()
+            if self.new_qsequence.count() > 4:
+                # QKeySequence accepts a maximum of 4 different sequences.
+                warning = SEQUENCE_LENGTH
+            elif self.new_sequence in BLACKLIST:
+                    warning = IN_BLACKLIST
+            elif self.check_shift_keyseq() and self.context != 'editor':
+                warning = SHIFT_BLACKLIST
+            elif self.check_singlekey() is False:
+                warning = INVALID_KEY
+            elif conflicts:
+                warning = SEQUENCE_CONFLICT
             else:
-                shown_sequence = sequence
-        else:
-            shown_sequence = sequence
-        self.text_new_sequence.setText(shown_sequence)
-        self.new_sequence = sequence
-
-        conflicts = self.check_conflicts()
-        blacklist = self.new_sequence in BLACKLIST
-        individual_keys = self.new_sequence.split('+')
-        if conflicts and different_sequence:
-            warning_type = SEQUENCE_CONFLICT
-        elif blacklist:
-            warning_type = IN_BLACKLIST
-        elif len(individual_keys) == 2 and individual_keys[0] == 'Shift':
-            warning_type = SHIFT_BLACKLIST
-        else:
-            warning_type = NO_WARNING
-
-        self.update_warning(warning_type=warning_type, conflicts=conflicts)
-
-    def validate_sequence(self):
-        """Provide additional checks for accepting or rejecting shortcuts."""
-        if self.invalid_key_flag:
-            self.update_warning(warning_type=INVALID_KEY)
-            return
-
-        for mod in MODIFIERS:
-            non_mod = set(self.key_non_modifiers)
-            non_mod.discard(mod)
-            if mod in self.key_non_modifiers:
-                self.key_non_modifiers.remove(mod)
-
-        self.key_modifiers = self.key_modifiers - non_mod
+                warning = NO_WARNING
 
-        while u'' in self.key_text:
-            self.key_text.remove(u'')
+            self.button_ok.setEnabled(warning == NO_WARNING)
+            self.update_warning(warning_type=warning, conflicts=conflicts)
 
-        self.key_text = [k.upper() for k in self.key_text]
+    def set_sequence_to_default(self):
+        """Set the new sequence to the default value defined in the config."""
+        default_keystr = QKeySequence(
+            CONF.get_default('shortcuts', '%s/%s' % (self.context, self.name))

```

Comment:
Any reason to not use the more modern and clear ``format()`` syntax here? Particularly since its what you use elsewhere...